src/pint/models/noise_model.py

```diff
@@ -27,6 +27,25 @@ def __init__(self,):
     def validate(self,):
         super(NoiseComponent, self).validate()
 
+    def validate_toas(self, toas, suppress_warning=False):
```

Comment:
This does something different from what `validate_toas` does in every other case: all other versions of the function raise an exception if any free parameter has no TOAs. This one doesn't care if the parameters are free, and doesn't raise an exception. If you still want a function that does what this one does, please call it something else.

Reply:
The original valiate_toa() will never raise any exception since the noise parameters are not get fitted currently. I understand the point to have only one behaviour for this type of functions. However, I think the noise model should warn the use about no TOA found, which is the goal of this function. I can give a different name, but in the end, if we want to warning the user about this, we will have to run the "new_name" function every time the fitting starts. Is there any risk to expend the behaviours  of validate_toa() functions

Reply:
On every other kind of object, `validate_toa` raises exceptions to indicate anything missing. Having a function of the same name do something different on this kind of object is confusing. Particularly as this is called through inheritance and by the timing model. All the other versions do nothing to warn of parameters with no TOAs unless they are being fit. Certainly they don't return anything. If you wanted all the `validate_toa` versions to warn about no-TOA-but-frozen parameters, and this one to begin raising exceptions if noise parameters became fittable, then we could have consistent behaviour. But this function is not necessary, as I say below, to fix the bug in question.

Reply:
Ok. I can remove this function here and I do think it is reasonable to give a warning when a frozen parameter has no toas. I got a lot of questions about the error massage caused by lacking of TOAs in the noise_design_matrix(). It is misleading.